part-2.py

```diff
@@ -4,13 +4,39 @@
 # the appropriate comment.
 
 # search
+def search(array, query):
+    # base case
+    if not array: 
+        return False
+    if array.pop() == query:
+        return True
 
+    return search(array[:-1], query)
```

Comment:
The Python list `pop` method returns the last element, but also removes it from the list.  The slice on line 14 will result in skipping an extra element each function call.